src/components/PeopleManagement/tests/DownloadCsvButton.test.jsx

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+import { IntlProvider } from '@edx/frontend-platform/i18n';
+import { logError } from '@edx/frontend-platform/logging';
+import { act, render, screen } from '@testing-library/react';
+
+import '@testing-library/jest-dom/extend-expect';
+
+import userEvent from '@testing-library/user-event';
+import DownloadCsvButton from '../DownloadCSVButton';
+import { downloadCsv } from '../../../utils';
+
+jest.mock('file-saver', () => ({
+  ...jest.requireActual('file-saver'),
+  saveAs: jest.fn(),
+}));
+
+jest.mock('../../../utils', () => ({
+  downloadCsv: jest.fn(),
+}));
+
+jest.mock('@edx/frontend-platform/logging', () => ({
+  ...jest.requireActual('@edx/frontend-platform/logging'),
+  logError: jest.fn(),
+}));
+
+jest.useFakeTimers({ advanceTimers: true }).setSystemTime(new Date('2024-01-20'));
+
+const mockData = {
+  results: [
+    {
+      enterprise_customer_user: {
+        email: 'a@letter.com',
+        joined_org: 'Apr 07, 2024',
+        name: 'A',
+      },
+      enrollments: 3,
+    },
+    {
+      enterprise_customer_user: {
+        email: 'b@letter.com',
+        joined_org: 'Apr 08, 2024',
+        name: 'B',
+      },
+      enrollments: 4,
+    },
+  ],
+};
+
+const testId = 'test-id-1';
+const DEFAULT_PROPS = {
+  totalCt: mockData.results.length,
+  fetchData: jest.fn(() => Promise.resolve(mockData)),
+  testId,
+};
+
+const DownloadCSVButtonWrapper = props => (
+  <IntlProvider locale="en">
+    <DownloadCsvButton {...props} />
+  </IntlProvider>
+);
+
+describe('DownloadCSVButton', () => {
+  const flushPromises = () => new Promise(setImmediate);
+
+  it('renders download csv button correctly.', async () => {
+    render(<DownloadCSVButtonWrapper {...DEFAULT_PROPS} />);
+    expect(screen.getByTestId(testId)).toBeInTheDocument();
+
+    // Validate button text
+    expect(screen.getByText('Download all (2)')).toBeInTheDocument();
+
+    // Click the download button.
+    screen.getByTestId(testId).click();
+    await flushPromises();
+
+    expect(DEFAULT_PROPS.fetchData).toHaveBeenCalled();
+    const expectedFileName = '2024-01-20-people-report.csv';
+    const expectedHeaders = ['Name', 'Email', 'Joined Organization', 'Enrollments'];
+    expect(downloadCsv).toHaveBeenCalledWith(expectedFileName, mockData.results, expectedHeaders, expect.any(Function));
+  });
+  it('download button should handle error returned by the API endpoint.', async () => {
```

Comment:
Can we add in a test for filtered results?

Reply:
Good point!  The DownloadCsvButton component doesn't have awareness of filters, and this would have to be tested at the level of the PeopleManagementPage.  As of right now there are no unit tests for the table part of the PeopleManagementPage, and adding that I feel is a substantial enough task outside the scope of this change to address it as a separate ticket.